src/exec/paths.rs

```diff
@@ -0,0 +1,117 @@
+pub fn basename(input: &str) -> &str {
+    let mut index = 0;
+    for (id, character) in input.bytes().enumerate() {
+        if character == b'/' {
```

Comment:
I think we should use something like `std::path::MAIN_SEPARATOR` here instead of `/`. This doesn't seem to work on Windows.

Reply:
Switched the module and it's tests to use `MAIN_SEPARATOR` and `char_indices`.